src/aliceVision/track/tracksUtils.hpp

```diff
@@ -0,0 +1,253 @@
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+#include <aliceVision/track/Track.hpp>
+
+
+namespace aliceVision {
+namespace track {
+
+using namespace aliceVision::matching;
+
+
+/**
+ * @brief Find common tracks between images.
+ * @param[in] imageIndexes: set of images we are looking for common tracks
+ * @param[in] tracksIn: all tracks of the scene
+ * @param[out] tracksOut: output with only the common tracks
+ */
+bool getCommonTracksInImages(const std::set<std::size_t>& imageIndexes,
+                                    const TracksMap& tracksIn,
+                                    TracksMap & tracksOut);
+  
+/**
+ * @brief Find common tracks among a set of images.
+ * @param[in] imageIndexes: set of images we are looking for common tracks.
+ * @param[in] tracksPerView: for each view it contains the list of visible tracks. *The tracks ids must be ordered*.
+ * @param[out] visibleTracks: output with only the common tracks.
+ */
+void getCommonTracksInImages(const std::set<std::size_t>& imageIndexes,
+                                    const TracksPerView& tracksPerView,
+                                    std::set<std::size_t>& visibleTracks);
+  
+/**
+ * @brief Find common tracks among images.
+ * @param[in] imageIndexes: set of images we are looking for common tracks.
+ * @param[in] tracksIn: all tracks of the scene.
+ * @param[in] tracksPerView: for each view the id of the visible tracks.
+ * @param[out] tracksOut: output with only the common tracks.
+ */
+bool getCommonTracksInImagesFast(const std::set<std::size_t>& imageIndexes,
+                                          const TracksMap& tracksIn,
+                                          const TracksPerView& tracksPerView,
+                                          TracksMap& tracksOut);
+  
+/**
+ * @brief Find all the visible tracks from a set of images.
+ * @param[in] imagesId set of images we are looking for tracks.
+ * @param[in] tracks all tracks of the scene.
+ * @param[out] tracksId the tracks in the images
+ */
+void getTracksInImages(const std::set<std::size_t>& imagesId,
+                              const TracksMap& tracks,
+                              std::set<std::size_t>& tracksId);
+
+/**
+ * @brief Find all the visible tracks from a set of images.
+ * @param[in] imagesId set of images we are looking for tracks.
+ * @param[in] tracksPerView for each view the id of the visible tracks.
+ * @param[out] tracksId the tracks in the images
+ */
+void getTracksInImagesFast(const std::set<IndexT>& imagesId,
+                                  const TracksPerView& tracksPerView,
+                                  std::set<IndexT>& tracksIds);
+
+/**
+ * @brief Find all the visible tracks from a single image.
+ * @param[in] imageIndex of the image we are looking for tracks.
+ * @param[in] tracks all tracks of the scene.
+ * @param[out] tracksIds the tracks in the image
+ */
+inline void getTracksInImage(const std::size_t& imageIndex,
```

Comment:
I'd move the implementation in the cop (and remove inline) for all the not-template functions